web/query.go

```diff
@@ -234,7 +240,11 @@ func (q queryHandler) ServeHTTP(writer http.ResponseWriter, request *http.Reques
 		}
 	default: // use the form parameters
 		if err := request.ParseForm(); err != nil {
-			writer.WriteHeader(http.StatusBadRequest)
+			if errHTTP, ok := err.(ErrorHTTP); ok {
```

Comment:
Fixed; it now does this only for the error coming from `q.process`, which does the real work.